src/main/java/com/odysseusinc/arachne/executionengine/service/versiondetector/DefaultVersionDetectionService.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *
+ * Copyright 2019 Odysseus Data Services, inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Company: Odysseus Data Services, Inc.
+ * Product Owner/Architecture: Gregory Klebanov
+ * Authors: Pavel Grafkin, Vitaly Koulakov, Anastasiia Klochkova, Yaroslav Molodkov, Alexander Cumarav
+ * Created: October 17, 2019
+ *
+ */
+
+package com.odysseusinc.arachne.executionengine.service.versiondetector;
+
+import com.odysseusinc.arachne.commons.types.CommonCDMVersionDTO;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.DataSourceUnsecuredDTO;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.join;
+
+@Service
+public class DefaultVersionDetectionService extends BaseVersionDetectionService implements VersionDetectionService {
+
+    private static final Logger log = LoggerFactory.getLogger(DefaultVersionDetectionService.class);
+
+    private final CDMSchemaProvider cdmSchemaProvider;
+    private final MetadataProvider metadataProvider;
+
+    @Inject
```

Comment:
why did you use `inject` over `autowire` annotation here?

Reply:
There is no difference. It's framework agnostic annotation, Spring supports it since 3.0. Do you see a problem here?

Reply:
Not a problem, but there's nice to have a single style across the whole team